drivers/dp/swdp_ll_pin.h

```diff
@@ -7,14 +7,18 @@
 #include <zephyr/kernel.h>
 #include <zephyr/drivers/gpio.h>
 #include <soc.h>
+#if defined(CONFIG_SOC_FAMILY_STM32)
+#include <stm32_ll_gpio.h>
+#endif
 
 #if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
 #define CPU_CLOCK		64000000U
 #else
 #define CPU_CLOCK		CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
 #endif
 
-#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X) || \
+	defined(CONFIG_SOC_FAMILY_STM32)
```

Comment:
@jfischer-no this strategy could get real messy real quick, want me to refactor this into multiple files? Like have an `swdp_ll_noop.h`, `swdp_ll_nrf.h`, `swdp_ll_stm32.h` and include them selectively from this fie.

Reply:
+1 creating architecture specific headers and then defining a macro/helper in each header to eliminate the #ifdefs in this file is much easier to read.

Reply:
done, way cleaner now